src/realm/query_expression.cpp

```diff
@@ -170,9 +177,8 @@ void LinkMap::map_links(size_t column, size_t row, LinkMapFunction& lm) const
             }
         }
     }
-    // Note: Link lists and link sets have compatible storage.
```

Comment:
Casting to an unrelated type is not legal even if we're pretty sure they happen to have member variables at the same places.